### ISSUE TITLE
[Linux] building obs-ndi "missing:  LIBOBS_LIB" 

### ISSUE THREAD

**Issue by walterav1984** [Palakis/obs-ndi#14]
Using the current obs-ndi build instructions I'm unable to build obs-ndi in ubuntu-1604 amd64. As far as I understand I have to get the current "obs-studio" source first and than follow your instructions from README.md?

```
mkdir mygitprojects
cd mygitprojects
git clone https://github.com/jp9000/obs-studio.git
```
Than I satisfied all build dependencies for obs-studio using the instructions from the following link (have tried with and without building it first).
https://github.com/jp9000/obs-studio/wiki/Install-Instructions#debian-based-build-directions

```
cd mygitprojects
git clone https://github.com/Palakis/obs-ndi.git
cd obs-ndi
git checkout 0.3.2 #also tried without 0.3.2
mkdir build && cd build
cmake -DLIBOBS_INCLUDE_DIR=/home/ubuntuadmin/mygitprojects/obs-studio/libobs -DNDISDK_DIR= /home/ubuntuadmin/Downloads/NDI\ SDK\ for\ Linux -DCMAKE_INSTALL_PREFIX=/usr ..

-- Could NOT find Libobs (missing:  LIBOBS_LIB) 
CMake Error at external/FindLibObs.cmake:106 (message):
  Could not find the libobs library
Call Stack (most recent call first):
  CMakeLists.txt:9 (include)

-- Configuring incomplete, errors occurred!
See also "/home/ubuntuadmin/mygitprojects/obs-ndi/build/CMakeFiles/CMakeOutput.log"
```
Also tried other path without "" or <> or both "<path>", also escape characters like "\ " for NDI location.

**Comment by walterav1984** [comment]
Adding "-DLIBOBS_LIB=home/ubuntuadmin/mygitprojects/obs-studio/libobs" before "-DLIBOBS_INCLUDE_DIR" brings me further.

```
cmake -DLIBOBS_LIB="/home/ubuntuadmin/mygitprojects/obs-studio/libobs" -DLIBOBS_INCLUDE_DIR="/home/ubuntuadmin/mygitprojects/obs-studio/libobs" -DNDISDK_DIR="/home/ubuntuadmin/Downloads/NDI SDK for Linux" -DCMAKE_INSTALL_PREFIX=/usr ..
-- Found Libobs: /home/ubuntuadmin/mygitprojects/obs-studio/libobs  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntuadmin/mygitprojects/obs-ndi/build
```

Building stills errors.

```make -j4
Scanning dependencies of target obs-ndi_automoc
[ 11%] Automatic moc and uic for target obs-ndi
Generating ui_output-settings.h
Generating moc_output-settings.cpp
[ 11%] Built target obs-ndi_automoc
Scanning dependencies of target obs-ndi
[ 22%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi.cpp.o
[ 33%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi-filter.cpp.o
[ 44%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi-output.cpp.o
[ 55%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi-source.cpp.o
[ 66%] Building CXX object CMakeFiles/obs-ndi.dir/Config.cpp.o
[ 77%] Building CXX object CMakeFiles/obs-ndi.dir/forms/output-settings.cpp.o
[ 88%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi_automoc.cpp.o
In file included from /home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:30:0:
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp: In function ‘void main_output_start(const char*)’:
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.h:43:74: warning: format ‘%s’ expects argument of type ‘char*’, but argument 3 has type ‘QString*’ [-Wformat=]
 #define blog(level, msg, ...) blog(level, "[obs-ndi] " msg, ##__VA_ARGS__)
                                                                          ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:136:3: note: in expansion of macro ‘blog’
   blog(LOG_INFO, "starting main NDI output with name '%s'", Config::Current()->
   ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp: In function ‘const char* GetNDILibPath()’:
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:194:8: warning: ISO C++ forbids converting a string constant to ‘char*’ [-Wwrite-strings]
   path = "/usr/lib/libndi.so.1.0.1";
        ^
[100%] Linking CXX shared module obs-ndi.so
/usr/bin/ld: cannot find -lobs-frontend-api
collect2: error: ld returned 1 exit status
CMakeFiles/obs-ndi.dir/build.make:254: recipe for target 'obs-ndi.so' failed
make[2]: *** [obs-ndi.so] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/obs-ndi.dir/all' failed
make[1]: *** [CMakeFiles/obs-ndi.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by walterav1984** [comment]
Building the 0.3.2 versions gives following error with building.

```
cmake -DLIBOBS_LIB="/home/ubuntuadmin/mygitprojects/obs-studio/libobs" -DLIBOBS_INCLUDE_DIR="/home/ubuntuadmin/mygitprojects/obs-studio/libobs" -DNDISDK_DIR="/home/ubuntuadmin/Downloads/NDISDKforLinux" -DCMAKE_INSTALL_PREFIX=/usr ..
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Libobs: /home/ubuntuadmin/mygitprojects/obs-studio/libobs  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ubuntuadmin/mygitprojects/obs-ndi/build
ubuntuadmin@q6600lin:~/mygitprojects/obs-ndi/build$ ls
CMakeCache.txt  CMakeFiles  cmake_install.cmake  Makefile
ubuntuadmin@q6600lin:~/mygitprojects/obs-ndi/build$ make -j4
Scanning dependencies of target obs-ndi
[ 16%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi.cpp.o
[ 33%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi-output.cpp.o
[ 50%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi-source.cpp.o
In file included from /usr/include/c++/5/cstdint:35:0,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.compat.h:26,
                 from /home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-output.cpp:24:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
[ 66%] Building CXX object CMakeFiles/obs-ndi.dir/obs-ndi-filter.cpp.o
In file included from /usr/include/c++/5/cstdint:35:0,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.compat.h:26,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.Lib.h:57,
                 from /home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:25:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
In file included from /usr/include/c++/5/cstdint:35:0,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.compat.h:26,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.Lib.h:57,
                 from /home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-source.cpp:26:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
In file included from /usr/include/c++/5/cstdint:35:0,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.compat.h:26,
                 from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.Lib.h:57,
                 from /home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-filter.cpp:28:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp: In function ‘bool obs_module_load()’:
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:75:2: warning: lambda expressions only available with -std=c++11 or -std=gnu++11
  }, nullptr);
  ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:75:5: error: ‘nullptr’ was not declared in this scope
  }, nullptr);
     ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:79:2: warning: lambda expressions only available with -std=c++11 or -std=gnu++11
  }, nullptr);
  ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:85:2: warning: lambda expressions only available with -std=c++11 or -std=gnu++11
  }, nullptr);
  ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp: In function ‘void main_output_start()’:
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi.cpp:105:82: error: ‘nullptr’ was not declared in this scope
 = obs_output_create("ndi_output", "main_ndi_output", output_settings, nullptr);
                                                                       ^
CMakeFiles/obs-ndi.dir/build.make:62: recipe for target 'CMakeFiles/obs-ndi.dir/obs-ndi.cpp.o' failed
make[2]: *** [CMakeFiles/obs-ndi.dir/obs-ndi.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/obs-ndi.dir/build.make:110: recipe for target 'CMakeFiles/obs-ndi.dir/obs-ndi-output.cpp.o' failed
make[2]: *** [CMakeFiles/obs-ndi.dir/obs-ndi-output.cpp.o] Error 1
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-source.cpp: In function ‘obs_properties_t* ndi_source_getproperties(void*)’:
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-source.cpp:59:2: warning: lambda expressions only available with -std=c++11 or -std=gnu++11
  });
  ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-source.cpp:71:2: warning: lambda expressions only available with -std=c++11 or -std=gnu++11
  });
  ^
/home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-source.cpp:74:72: error: invalid conversion from ‘int*’ to ‘uint32_t* {aka unsigned int*}’ [-fpermissive]
  s->ndi_sources = NDIlib_find_get_sources(ndi_finder, &s->no_sources, 0);
                                                                        ^
In file included from /home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.Lib.h:84:0,
                 from /home/ubuntuadmin/mygitprojects/obs-ndi/obs-ndi-source.cpp:26:
/home/ubuntuadmin/Downloads/NDISDKforLinux/include/Processing.NDI.Find.h:78:24: note:   initializing argument 2 of ‘const NDIlib_source_t* NDIlib_find_get_sources(NDIlib_find_instance_t, uint32_t*, uint32_t)’
 const NDIlib_source_t* NDIlib_find_get_sources(NDIlib_find_instance_t p_instance, uint32_t* p_no_sources, uint32_t timeout_in_ms);
                        ^
CMakeFiles/obs-ndi.dir/build.make:86: recipe for target 'CMakeFiles/obs-ndi.dir/obs-ndi-source.cpp.o' failed
make[2]: *** [CMakeFiles/obs-ndi.dir/obs-ndi-source.cpp.o] Error 1
CMakeFiles/obs-ndi.dir/build.make:134: recipe for target 'CMakeFiles/obs-ndi.dir/obs-ndi-filter.cpp.o' failed
make[2]: *** [CMakeFiles/obs-ndi.dir/obs-ndi-filter.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/obs-ndi.dir/all' failed
make[1]: *** [CMakeFiles/obs-ndi.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2

```

**Comment by Palakis** [comment]
First of all, you need to build and install obs-studio. This will create the necessary library the plugin will link to.

What is your G++ version number ?

**Comment by Palakis** [comment]
Ah, GCC 5.4.0.

**Comment by Palakis** [comment]
After building and installing obs-studio, remove the LIBOBS_LIB variable. It will be auto-detected.

**Comment by walterav1984** [comment]
It was indeed needed to also "install" obs-studio instead of building it only like you mentioned, than removing the "-DLIBOBS_LIB" variable I added myself in the cmake command but most important is to use newest git version of "obs-ndi" and not checkout 0.3.2 (since you fixed allot of linux stuff lately).

Building succeeds, but running obs popped up the "NDI redistributable download message"(lol windows only exe, shame on you Newtek ;-) since I installed the "libndi.so.1.0.1" in /usr/local/ndi/lib and it was not found by obs because it is hardcoded in "obs-ndi.cpp".

obs-ndi.cpp:
```
		path = "/usr/lib/libndi.so.1.0.1";
```

Copying the libndi.so.1.0.1 /usr/lib/libndi.so.1.0.1 or changing the path before compiling fixed running "obs-ndi" for on Ubuntu 16.04 amd64 linux.

Thank you!
